framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4631,6 +4619,452 @@ private void makeGroundTargetType(
+      AnnotatedTypeMirror type, TypeMirror typeMirror) {
+    if (!shouldCapture(type, typeMirror)) {
+      if (type.containsUninferredTypeArguments()
+          && typeMirror.getKind() == TypeKind.DECLARED
+          && type.getKind() == TypeKind.DECLARED) {
+        // If capture conversion is not applied because the type contains uninferred
+        // type arguments, then mark all wildcards in type as "uninferred" before it is returned.
+        DeclaredType capturedTypeMirror = (DeclaredType) typeMirror;
+        AnnotatedDeclaredType uncapturedType = (AnnotatedDeclaredType) type;
+        for (int i = 0; i < capturedTypeMirror.getTypeArguments().size(); i++) {
+          TypeMirror capturedTypeArgTM = capturedTypeMirror.getTypeArguments().get(i);
+          AnnotatedTypeMirror uncapturedTypeArg = uncapturedType.getTypeArguments().get(i);
+          if (uncapturedTypeArg.getKind() != TypeKind.WILDCARD) {
+            continue;
+          }
+          if (TypesUtils.isCapturedTypeVariable(capturedTypeArgTM)
+              || capturedTypeArgTM.getKind() != TypeKind.WILDCARD) {
+            ((AnnotatedWildcardType) uncapturedTypeArg).setUninferredTypeArgument();
+          }
+        }
+      }
+      return type;
+    }
+
+    AnnotatedDeclaredType uncapturedType = (AnnotatedDeclaredType) type;
+    DeclaredType capturedTypeMirror = (DeclaredType) typeMirror;
+    AnnotatedDeclaredType capturedType =
+        (AnnotatedDeclaredType) AnnotatedTypeMirror.createType(capturedTypeMirror, this, false);
+
+    captureTypeArgCopier.copy(uncapturedType, capturedType);
+
+    AnnotatedDeclaredType typeDeclaration =
+        (AnnotatedDeclaredType) getAnnotatedType(uncapturedType.getUnderlyingType().asElement());
+
+    // A mapping from type variable to its type argument in the captured type.
+    Map<TypeVariable, AnnotatedTypeMirror> typeVariableMapping = new HashMap<>();
+    // A mapping from a captured type variable to the annotated captured type variable.
+    Map<TypeVariable, AnnotatedTypeVariable> capturedTypeVarMapping = new HashMap<>();
+    // `newTypeArgs` will be the type arguments of the result of this method.
+    List<AnnotatedTypeMirror> newTypeArgs = new ArrayList<>();
+    for (int i = 0; i < typeDeclaration.getTypeArguments().size(); i++) {
+      TypeVariable typeVarTypeMirror =
+          (TypeVariable) typeDeclaration.getTypeArguments().get(i).getUnderlyingType();
+      AnnotatedTypeMirror uncapturedTypeArg = uncapturedType.getTypeArguments().get(i);
+      AnnotatedTypeMirror capturedTypeArg = capturedType.getTypeArguments().get(i);
+      if (uncapturedTypeArg.getKind() == TypeKind.WILDCARD) {
+        if (TypesUtils.isCapturedTypeVariable(capturedTypeArg.getUnderlyingType())) {
+          // The type argument is a captured type variable. Use the type argument from the newly
+          // created and yet-to-be annotated capturedType. (The annotations are added as
+          // part of capturing the wildcard.)
```

Comment:
"capturing the wildcard" is funny terminology; consider rewording, per our earlier discussion.

Also, where does the "capturing the wildcard" occur?  It would be helpful for this comment to be more explicit.